tests/rbd_mirror/test-image-delete-primary-site.py

```diff
@@ -29,33 +50,20 @@ def run(**kw):
     2. Delete the image when contents are getting mirrored.
     3. Make sure that image is deleted in secondary site.
     """
-    try:
-        log.info("Starting RBD mirroring test case - 9501")
-        config = kw.get("config")
-        mirror1, mirror2 = [
-            rbdmirror.RbdMirror(cluster, config)
-            for cluster in kw.get("ceph_cluster_dict").values()
-        ]
-        poolname = mirror1.random_string() + "_ceph_9501"
-        imagename = mirror1.random_string() + "_ceph_9501"
-        imagespec = poolname + "/" + imagename
-
-        mirror1.initial_mirror_config(
-            mirror2,
-            poolname=poolname,
-            imagename=imagename,
-            imagesize=config.get("imagesize", "1G"),
-            mode="pool",
-            **kw,
-        )
-
-        mirror1.benchwrite(imagespec=imagespec, io=config.get("io-total", "1G"))
+    log.info("Starting RBD mirroring test case - 9501")
+    mirror_obj = rbd_mirror_config(**kw)
 
-        mirror1.delete_image(imagespec)
-        if mirror2.image_exists(imagespec):
-            return 0
+    if mirror_obj:
+        log.info("Executing test on replicated pool")
+        if test_image_delete_from_primary(
+            mirror_obj.get("rep_rbdmirror"), "rep_pool_config", **kw
+        ):
+            return 1
 
-    except Exception as e:
-        log.exception(e)
+        log.info("Executing test on ec pool")
+        if test_image_delete_from_primary(
+            mirror_obj.get("ec_rbdmirror"), "ec_pool_config", **kw
+        ):
+            return 1
 
-    return 1
+    return 0
```

Comment:
if `mirror_obj` empty still it's returing 0 so better add `return 0` inside if and `return 1` outside if

Reply:
In team review we thought of taking that scenario as common for all methods hence resolving it now.